tests/unit/components/lg-dob-search-selector-test.js

```diff
@@ -0,0 +1,91 @@
+import Ember from 'ember';
+import { dobSearchOptions } from 'lion-guardians/utils/units';
+
+import {
+  moduleForComponent,
+  test
+} from 'ember-qunit';
+
+moduleForComponent('lg-dob-search-selector', 'LgDobSearchSelectorComponent', {
+  setup: function() {
+    this.container.register('view:select', Ember.Select);
+  }
+});
+
+test('it renders', function() {
+  expect(2);
+
+  // creates the component instance
+  var component = this.subject();
+  equal(component._state, 'preRender');
+
+  // appends the component to the page
+  this.append();
+  equal(component._state, 'inDOM');
+});
+
+test('it does not set start or end when no age selected', function(){
+  expect(2);
+  var component = this.subject();
+
+  component.set('selectedDobOption', null);
+  equal(component.get('selectedSearchDateEnd'), null);
+  equal(component.get('selectedSearchDateStart'), null);
+});
+
+test('it sets selected start and end for age for very young lions', function(){
+  expect(3);
+  var component = this.subject();
+  var selectedOption = dobSearchOptions[1].value;
+
+  var today = new Date();
+
+  var startDate = new Date();
+  startDate.setFullYear(today.getFullYear() - selectedOption.maximumAge);
+
+  var endDate = new Date(); //
+
+  component.set('selectedDobOption', selectedOption);
+  equal(component.get('selectedDobOption'), selectedOption);
+  equal(component.get('selectedSearchDateStart'), startDate);
+  equal(component.get('selectedSearchDateEnd'), endDate);
+});
+
+test('it sets params for age for very old lions', function(){
+  expect(3);
+  var component = this.subject();
+  var numOptions = dobSearchOptions.length;
+  var selectedOption = dobSearchOptions[numOptions - 1].value;
+
+  var today = new Date();
+
+  var startDate = new Date();
+  startDate.setFullYear(today.getFullYear() - 100);
+
+  var endDate = new Date(); //
+  endDate.setFullYear(today.getFullYear() - selectedOption.minimumAge);
+
+  component.set('selectedDobOption', selectedOption);
+  equal(component.get('selectedDobOption'), selectedOption);
```

Comment:
It may be late in the game to suggest this, but considering that we are using year-based ranges, I'm not sure we really even need to send dates across.
The very simplest thing would be to push this complexity to the server — it should understand how to search for a "3 year-old lion +/- 1 year". You can send the server a query like {age: 3, range: 1} to indicate 2 to 4 years, and let the server worry about how to translate that into an appropriate SQL query.

Nothing intrinsically wrong with working with date objects client-side, but an age + range query may map better to the search domain.